modules/geo-layers/src/tile-layer/tile-layer.ts

```diff
@@ -235,7 +243,7 @@ export default class TileLayer<DataT = any, ExtraPropsT extends {} = {}> extends
   }
 
   private _updateTileset(): void {
-    const {tileset} = this.state;
+    const tileset = this.state.tileset!;
```

Comment:
Can we make tileset non-optional in state so we don't have to override type system everywhere? Can a tile layer be constructed without a tileset?

Reply:
I think this is a good discussion to have, though refactoring might be better to do in another PR. My hope is for these audits to generate discussion like this. I've made #8408 to follow up.